esx_service/utils/auth_api.py

```diff
@@ -365,11 +365,68 @@ def is_vm_duplicate(vm_list):
 
     return None
 
+def check_default_datastore(datastore_name):
+    """ Check datastore with given name is a valid value for default_datastore """
+    # The valid default_datastore name are:
+    # named datastore existing on the host
+    # hard coded datastore name "__VM_DS"
+    # "__ALL_DS" is not a valid value to set as "default_datastore"
+    if datastore_name == auth_data_const.VM_DS:
+        return None
+    if datastore_name == auth_data_const.ALL_DS:
+        return error_code.generate_error_info(ErrorCode.DEFAULT_DS_NAME_INVALID, datastore_name)
+
+    if not vmdk_utils.validate_datastore(datastore_name):
+        error_info = error_code.generate_error_info(ErrorCode.DS_NOT_EXIST, datastore_name)
+        return error_info
+
+    return None
+
+def set_default_ds_and_create_privilege(tenant, default_datastore, check_existing):
```

Comment:
set_default_ds() is good enough. There is no need to describe all details in the name, 
especially since you put the correct info in docstring